pkg/rbd/nodeserver.go

```diff
@@ -71,18 +72,21 @@ func (ns *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 		return &csi.NodePublishVolumeResponse{}, nil
 	}
 
-	// if our access mode is a simple SINGLE_NODE_WRITER we're going to ignore the SC directive and use the
-	// watcher still
-	ignoreMultiWriterEnabled := true
-	if req.VolumeCapability.AccessMode.Mode != csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER {
-		ignoreMultiWriterEnabled = false
+	// MULTI_NODE_MULTI_WRITER is supported by default for Block access type volumes
+	if req.VolumeCapability.AccessMode.Mode == csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER {
+		if isBlock {
+			disableInUseChecks = true
+		} else {
+			klog.Warningf("MULTI_NODE_MULTI_WRITER currently only supported with volumes of access type `block`, invalid AccessMode for volume: %v", req.GetVolumeId())
+			e := fmt.Errorf("rbd: MULTI_NODE_MULTI_WRITER access mode only allowed with BLOCK access type")
+			return nil, status.Error(codes.InvalidArgument, e.Error())
```

Comment:
we can just return `status.Error(codes.InvalidArgument, "rbd: MULTI_NODE_MULTI_WRITER access mode only allowed with BLOCK access type")` instead of creating a new variable `e`

Reply:
sure, changed